CMakeLists.txt

```diff
@@ -665,13 +669,16 @@ add_custom_target(check
     DEPENDS check-recursive-init check-install-via-properties check-install-via-properties-with-debug check-junit)
 
 if(ENABLE_NATIVE_TEST_HOOKS)
-    add_custom_target(check-rdrand
-        COMMAND $<TARGET_FILE:test_rdrand>
-    )
+    if (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+      add_custom_target(check-rdrand
+            COMMAND $<TARGET_FILE:test_rdrand>
+        )
+      add_dependencies(check check-rdrand)
```

Comment:
Does this mean you are adding `check` twice?  Here and below?

Reply:
No. The first element `check` is getting a new dependency `check-rdrand`.